src/components/LinksPanel/components/ChatDrawer/ChatDrawer.tsx

```diff
@@ -0,0 +1,520 @@
+                            )}
+                          </div>
+                          <IconButton
+                            name="times"
+                            aria-label={`Remove ${file.name}`}
+                            onClick={() => removeAttachedFile(file.id)}
+                            size="sm"
+                            variant="secondary"
+                            className={styles.removeButton}
+                            {...testIds.removeButton.apply()}
+                          />
+                        </div>
+                      ) : null
+                    )}
+                  </div>
+                </div>
+              )}
+
+              <div className={styles.inputArea}>
+                <div className={styles.textareaContainer}>
+                  <TextArea
+                    ref={textareaRef}
+                    value={inputValue}
+                    onChange={handleInputChange}
+                    onKeyDown={handleKeyDown}
+                    placeholder="Type your message..."
+                    disabled={isLoading}
+                    className={styles.textarea}
+                    {...testIds.input.apply()}
+                  />
+
+                  <div className={styles.buttonsContainer}>
+                    <IconButton
+                      name="upload"
+                      aria-label="Attach files (images, documents, text files)"
+                      onClick={() => fileInputRef.current?.click()}
+                      disabled={isLoading}
+                      className={styles.attachButton}
+                      title="Attach files (images, documents, text files)"
+                      {...testIds.attachButton.apply()}
+                    />
+
+                    <input
+                      ref={fileInputRef}
+                      type="file"
+                      multiple
+                      accept="image/*,.txt,.pdf,.doc,.docx,.json,.csv"
+                      onChange={onFileInputChange}
+                      style={{ display: 'none' }}
+                      {...testIds.fileInput.apply()}
+                    />
+
+                    {isLoading ? (
+                      <IconButton
+                        name="fa fa-spinner"
+                        aria-label="Send message (Ctrl+Enter)"
+                        onClick={handleSend}
+                        disabled={(!inputValue.trim() && attachedFiles.length === 0) || isLoading}
+                        className={styles.sendButton}
+                        title="Send message (Ctrl+Enter)"
+                        {...testIds.sendButton.apply()}
+                      />
+                    ) : (
+                      <IconButton
+                        name="message"
+                        aria-label="Send message (Ctrl+Enter)"
+                        onClick={handleSend}
+                        disabled={(!inputValue.trim() && attachedFiles.length === 0) || isLoading}
```

Comment:
It's a duplication of 495 line. please create a variable `isSendDisabled` and reuse it